src/API/Common/Event/Dispatcher.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\API\Common\Event;
+
+use CloudEvents\V1\CloudEventInterface;
+use OpenTelemetry\Context\Context;
+use OpenTelemetry\Context\ContextKey;
+
+class Dispatcher
+{
+    private static ?ContextKey $key = null;
+    /** @var array<string, array<int, array<callable>>> */
+    private array $listeners = [];
+
+    private function __construct()
```

Comment:
Please, please do not make this a singleton. If we need a static reference to an instance, we already have it with the method below, and there are other ways to solve this in the future.
In fact I can think of a lot of places where an additional dispatcher would make sense.

Reply:
done